systest/online-restore/namespace-aware/restore_test.go

```diff
@@ -0,0 +1,251 @@
+	defer cleanup()
+
+	require.NoError(t, gc.Login(context.Background(), "groot", "password"))
+
+	query := `{
+		allMails(func: has(email)) {
+			   count(uid)
+			 }
+	 }`
+	resp, err := gc.Query(query)
+	require.NoError(t, err)
+	// there should be 100 emails in DB.
+	require.NoError(t, dgraphtest.CompareJSON(`{"allMails":[{"count":100}]}`, string(resp.Json)))
+
+	query = `{
+		allMails(func: has(name)) {
+			   count(uid)
+			 }
+	 }`
+	resp, err = gc.Query(query)
+	require.NoError(t, err)
+	require.NoError(t, dgraphtest.CompareJSON(`{"allMails":[{"count":0}]}`, string(resp.Json)))
+}
+
+func TestNamespaceAwareRestoreOnMultipleGroup(t *testing.T) {
+	conf := dgraphtest.NewClusterConfig().WithNumAlphas(6).WithNumZeros(6).
+		WithReplicas(3).WithACL(20 * time.Hour).WithEncryption().WithUidLease(10000)
+	c, err := dgraphtest.NewLocalCluster(conf)
+	x.Panic(err)
+	if err := c.Start(); err != nil {
+		c.Cleanup(true)
+	}
+
+	hc, err := c.HTTPClient()
+	require.NoError(t, err)
+	require.NoError(t, hc.LoginIntoNamespace("groot", "password", 0))
+	gc, cleanup, err := c.Client()
+	require.NoError(t, err)
+	defer cleanup()
+	require.NoError(t, gc.Login(context.Background(), "groot", "password"))
+
+	ns1, err := hc.AddNamespace()
+	require.NoError(t, err)
+	fmt.Println("Created namespace ", ns1)
+	require.NoError(t, gc.LoginIntoNamespace(context.Background(), "groot", "password", ns1))
+
+	require.NoError(t, gc.SetupSchema(emailSchema))
+	rdf := ""
+	for i := 1; i <= 100; i++ {
+		rdf = rdf + fmt.Sprintf("_:a%v <email> \"example%v@gmail.com\" .	\n", i, i)
+
+	}
+	_, err = gc.Mutate(&api.Mutation{
+		SetNquads: []byte(rdf),
+		CommitNow: true,
+	})
+	require.NoError(t, err)
+
+	ns2, err := hc.AddNamespace()
+	require.NoError(t, err)
+	fmt.Println("Created namespace ", ns2)
+
+	require.NoError(t, gc.LoginIntoNamespace(context.Background(), "groot", "password", ns2))
+	require.NoError(t, gc.SetupSchema(emailSchema))
+	rdf = ""
+	for i := 1; i <= 300; i++ {
+		rdf = rdf + fmt.Sprintf("_:a%v <email> \"example%v@gmail.com\" .	\n", i, i)
+
+	}
+	_, err = gc.Mutate(&api.Mutation{
+		SetNquads: []byte(rdf),
+		CommitNow: true,
+	})
+	require.NoError(t, err)
+
+	require.NoError(t, gc.LoginIntoNamespace(context.Background(), "groot", "password", 0))
+	require.NoError(t, gc.SetupSchema(emailSchema))
+
+	rdf = ""
+	for i := 1; i < 300; i++ {
+		rdf = rdf + fmt.Sprintf("_:a%v <email> \"example%v@gmail.com\" .	\n", i, i)
+		_, _ = gc.Mutate(&api.Mutation{
+			SetNquads: []byte(rdf),
+			CommitNow: true,
+		})
+	}
+	_, err = gc.Mutate(&api.Mutation{
+		SetNquads: []byte(rdf),
+		CommitNow: true,
+	})
+	require.NoError(t, err)
+
+	fmt.Println("Inserted email predicate into namespace 0")
+
+	require.NoError(t, gc.Login(context.Background(), "groot", "password"))
+	require.NoError(t, hc.LoginIntoNamespace("groot", "password", 0))
+
+	// creating multiple namespaces and adding email predicate into them
+	// so email predicte distributed into both groups
```

Comment:
can you write down in comments what is the distribution of predicates in the groups at this point?